tensorboard/plugins/debugger_v2/debug_data_multiplexer.py

```diff
@@ -36,9 +37,41 @@
 # ones are either repetitions of the earlier ones or caused by the earlier ones.
 DEFAULT_PER_TYPE_ALERT_LIMIT = 1000
 
+# Default interval between successive calls to `DebugDataReader.update()``.
+DEFAULT_RELOAD_INTERVAL_SEC = 60
 
-def run_in_background(target):
-    """Run a target task in the background.
+
+class Timer(object):
+    """A timer that supports interruptible sleep()."""
+
+    def __init__(self, interval_sec):
+        """Constructor for Timer.
+
+        Args:
+          interval_sec: Amount of time to sleep on each subsequent `sleep()`
+            call (assuming not waken by `waken()`), in seconds.
+        """
+        self._interval_sec = interval_sec
+        self._event = threading.Event()
+
+    def sleep(self):
+        """Sleep until `interval_sec` is reached or waken by `wake()`."""
+        self._event.clear()
+        self._event.wait(timeout=self._interval_sec)
+
+    def wake(self):
+        """Interrupt all sleep() calls that are ongoing.
+
+        This is a no-op if no sleep() is ongoing.
+        """
+        self._event.set()
+
+
+_timer = Timer(DEFAULT_RELOAD_INTERVAL_SEC)
+
+
+def run_repeatedly_in_background(target):
```

Comment:
FWIW, I think a somewhat nicer API for this function would be:

- take the interval at which to repeatedly call the target as a parameter to pass into `sleep()`, rather than making it a hidden global config that this function happens to use
- create the `threading.Event` as a local variable within this function and have `_run_repeatedly()` close over it
- return the `Event` as the return value to the caller, to be used by the caller to signal when `target` should run immediately

Reply:
Your suggestion here makes sense. Adopted. Unit tests are updated accordingly. Thanks.